tests/common/platform/interface_utils.py

```diff
@@ -109,24 +103,33 @@ def check_interface_information(dut, asic_index, interfaces, xcvr_skip_list):
 
     return True
 
+
 def get_port_map(dut, asic_index=None):
     """
     @summary: Get the port mapping info from the DUT
     @return: a dictionary containing the port map
     """
     logging.info("Retrieving port mapping from DUT")
-    # copy the helper to DUT
-    src_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'files/getportmap.py')
-    dest_path = os.path.join('/usr/share/sonic/device', dut.facts['platform'], 'plugins/getportmap.py')
-    dut.copy(src=src_path, dest=dest_path)
-
-    # execute command on the DUT to get portmap
-    get_portmap_cmd = "docker exec pmon python /usr/share/sonic/platform/plugins/getportmap.py -asicid {}".format(asic_index)
-    portmap_json_string = dut.command(get_portmap_cmd)["stdout"]
-
+    if asic_index is not None:
+        asichost = dut.asic_instance(asic_index)
+        cfg_facts = asichost.config_facts(host=dut.hostname, source="persistent", verbose=False)['ansible_facts']
+    else:
+        # execute command on the DUT to get port map from config_db.json
+        validate_config_db_file_exist(dut)
+        get_cfg_facts_cmd = "cat /etc/sonic/config_db.json"
+        portmap_json_string = dut.command(get_cfg_facts_cmd)["stdout"]
+        cfg_facts = json.loads(portmap_json_string)
     # parse the json
-    port_mapping = json.loads(portmap_json_string)
-    assert port_mapping, "Retrieve port mapping from DUT failed"
-
-    return port_mapping
-
+    port_mapping_res = {}
+    port_mapping = cfg_facts.get("PORT", {})
+    for port, port_dict_info in port_mapping.items():
+        port_index = port_dict_info["index"]
+        port_mapping_res[port] = [int(port_index)]
+    return port_mapping_res
+
+
+def validate_config_db_file_exist(dut):
+    config_db_location_cmd = "ls /etc/sonic/config_db.json"
+    config_db_location_string = dut.command(config_db_location_cmd)["stdout"]
```

Comment:
Can you use the ansible `state` module to check the existence of a file? It is much easier and more robust than running shell command and searching for specific string in output. Ref: https://docs.ansible.com/ansible/latest/collections/ansible/builtin/stat_module.html

Reply:
@wangxin fixed